src/app/features/cra/cra.component.ts

```diff
@@ -1,10 +1,287 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { CalendarOptions, DateSelectArg } from '@fullcalendar/core';
+import frLocale from '@fullcalendar/core/locales/fr';
+import dayGridPlugin from '@fullcalendar/daygrid';
+import interactionPlugin from '@fullcalendar/interaction';
+import listPlugin from '@fullcalendar/list';
+import timeGridPlugin from '@fullcalendar/timegrid';
+import { format, isWeekend } from 'date-fns';
+import { ErrorHandlerService } from 'src/app/core/template/components/error-dialog/error-handler.service';
+import { ModalComponent } from 'src/app/core/template/components/modal/modal.component';
+import { SucessHandlerService } from 'src/app/core/template/components/success-dialog/success-dialog.service';
+import { CraService } from '../../core/service/cra-service';
+import { CraModel } from 'src/app/core/model/cra-model';
 
 @Component({
   selector: 'app-cra',
   templateUrl: './cra.component.html',
-  styleUrls: ['./cra.component.scss']
+  styleUrls: ['./cra.component.scss'],
 })
-export class CraComponent {
+export class CraComponent implements OnInit {
+  currentYear = new Date().getFullYear();
+  HOLIDAYS = [
```

Comment:
its better to have a consts class where you export all app consts

Reply:
tu as raison une idée comment bien structurer tout ça de maniére propre ?